gnucash/gnome-utils/dialog-utils.c

```diff
@@ -876,3 +879,192 @@ gnc_get_negative_color (void)
+        {
+            gchar *guid_str = scm_to_utf8_string (SCM_CAR(rpt_guids));
+            gchar *name = gnc_scm_to_utf8_string (scm_call_2(template_menu_name,
+                                                  SCM_CAR(rpt_guids), SCM_BOOL_F));
+
+            gtk_list_store_append (store, &iter);
+            gtk_list_store_set (store, &iter,
+                                COL_INV_NAME, name,
+                                COL_INV_GUID, guid_str,
+                                COL_INV_MISSING, FALSE,
+                                -1);
+            g_free (name);
+            g_free (guid_str);
+
+            rpt_guids = SCM_CDR(rpt_guids);
+        }
+    }
+}
+
+static gboolean
+select_default (GtkWidget *combo, const gchar *default_guid)
+{
+    GtkTreeModel *model = gtk_combo_box_get_model (GTK_COMBO_BOX(combo));
+    GtkTreeIter iter;
+    gboolean found = FALSE;
+    gboolean valid_iter = gtk_tree_model_get_iter_first (model, &iter);
+
+    while (valid_iter)
+    {
+        gchar *guid;
+        gtk_tree_model_get (model, &iter, COL_INV_GUID, &guid, -1);
+
+        if (g_strcmp0 (default_guid, guid) == 0)
+        {
+            gtk_combo_box_set_active_iter (GTK_COMBO_BOX(combo), &iter);
+            g_free (guid);
+            found = TRUE;
+            break;
+        }
+        g_free (guid);
+        valid_iter = gtk_tree_model_iter_next (model, &iter);
+    }
+    return found;
+}
+
+static void
+combo_changed_cb (GtkComboBox *widget, gpointer user_data)
+{
+    GtkTreeIter iter;
+
+    if (gtk_combo_box_get_active_iter (widget, &iter))
+    {
+        QofBook *book = gnc_get_current_book ();
+        GtkTreeModel *model = gtk_combo_box_get_model (widget);
+        gchar *name, *guid;
+        gboolean missing;
+        gtk_tree_model_get (model, &iter, COL_INV_NAME, &name,
+                                          COL_INV_GUID, &guid,
+                                          COL_INV_MISSING, &missing,
+                                          -1);
+        // set visibility of the warning image
+        gtk_widget_set_visible (user_data, missing);
+
+        gncInvoiceSetDefaultReport (book, guid, name);
+
+        gtk_widget_queue_resize (GTK_WIDGET(widget));
+        g_free (name);
+        g_free (guid);
+    }
+}
+
+void
+gnc_default_print_report_list (GtkWidget *combo, GtkWidget *warning)
+{
+    GtkTreeModel *model = gtk_combo_box_get_model (GTK_COMBO_BOX(combo));
+    QofBook *book = gnc_get_current_book ();
+    const gchar *default_guid = gnc_get_default_invoice_print_report ();
+    const gchar *default_name = gncInvoiceGetDefaultReportName (book);
+    gboolean have_default = FALSE;
+
+    update_invoice_list (GTK_LIST_STORE(model));
+    have_default = select_default (combo, default_guid);
+
+    if (!have_default)
+    {
+        GtkTreeIter iter;
+        /* Translators: %s is the default invoice report name. */
+        gchar *tool_tip = g_strdup_printf (_("'%s' is missing"),
```

Comment:
Full stop?

HIG and Weblate like grammatical correct sentences.

Reply:
Thanks, missed that, I also have a post-it note to remind me!
Will change.